src/browser/speaker.js

```diff
@@ -19,145 +22,1121 @@ function SpeakerAdapter(bus)
+                    ];
+
+                    // Copy the first, already-shifted, small buffer into the new buffer.
+                    new_big_buffer[0].set(this.source_buffer_current[0]);
+                    new_big_buffer[1].set(this.source_buffer_current[1]);
+                    var new_big_buffer_pos = this.source_buffer_current[0].length;
+
+                    // Copy the rest.
+                    for(var i = 0; i < buffer_count; i++)
+                    {
+                        var small_buffer = this.queue_shift();
+                        new_big_buffer[0].set(small_buffer[0], new_big_buffer_pos);
+                        new_big_buffer[1].set(small_buffer[1], new_big_buffer_pos);
+                        new_big_buffer_pos += small_buffer[0].length;
+                    }
+
+                    // Pretend that everything's just fine.
+                    this.source_buffer_current = new_big_buffer;
+                }
+
+                this.pump();
+            }
+
+            pump()
+            {
+                if(this.queued_samples / this.source_samples_per_destination < QUEUE_RESERVE)
+                {
+                    this.port.postMessage(
+                    {
+                        type: "pump"
+                    });
+                }
+            }
+
+            queue_push(item)
+            {
+                if(this.queue_length < this.queue_size)
+                {
+                    this.queue_data[this.queue_end] = item;
+                    this.queue_end = this.queue_end + 1 & this.queue_size - 1;
+                    this.queue_length++;
+
+                    this.queued_samples += item[0].length;
+
+                    this.pump();
+                }
+            }
+
+            queue_shift()
+            {
+                if(!this.queue_length)
+                {
+                    return create_empty_buffer();
+                }
+
+                var item = this.queue_data[this.queue_start];
+
+                this.queue_data[this.queue_start] = null;
+                this.queue_start = this.queue_start + 1 & this.queue_size - 1;
+                this.queue_length--;
+
+                this.queued_samples -= item[0].length;
+
+                return item;
+            }
+
+            dbg_log(message)
+            {
+                this.port.postMessage(
+                {
+                    type: "debug-log",
+                    value: message
+                });
+            }
+        }
+
+        registerProcessor("dac-processor", DACProcessor);
+    }`;
+
+    //var worklet_string = worklet.toString();
+
+    var worklet_code_start = worklet_string.indexOf("{") + 1;
+    var worklet_code_end = worklet_string.lastIndexOf("}");
+    var worklet_code = worklet_string.substring(worklet_code_start, worklet_code_end);
+
+    var worklet_blob = new Blob([worklet_code], { type: "application/javascript" });
+    var worklet_url = URL.createObjectURL(worklet_blob);
```

Comment:
I think this has a *super-minor* memory leak in that the worklet url is never [`revoked`](https://developer.mozilla.org/en-US/docs/Web/API/URL/revokeObjectURL). Can we safely revoke it in the `.then` block?